src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -1348,7 +1354,7 @@ void dcd_int_handler(uint8_t rhport)
     do
     {
       handle_rxflvl_irq(rhport);
-    } while(dwc2->gotgint & GINTSTS_RXFLVL);
+    } while(dwc2->gintsts & GINTSTS_RXFLVL);
```

Comment:
This change is kinda confusing, howcome it was like this before? Is this change valid for gd32 & stm32?

Reply:
That was a bug, likely because of a typo. The goal here is to remain in the FIFO handler until the FIFO is empty and the interrupt is then automatically cleared again. Before the handler would be triggered again until the FIFO is empty instead of staying there until done as intended.

_During a code check I noticed that this was off so I looked closer. Checking the global OTG interrupt flag register (`gotgint`) isn't related to the FIFO so using that don't make sense, the naming of the bit also indicate that it's not supposed to be used with that register and a quick check on the current TinyUSB branch revealed that they also changed it._